server/gendb/table.go.tmpl

```diff
@@ -513,12 +553,26 @@ func (t {{ $modelName}}Table) get{{ $modelName }}sBy{{ $computedIndexName }}(ctx
     }
   }
 
+  {{- if gt $lenNonIndexAttributes 0 }}
+    if len(input.FilterValues) > 0 && input.FilterExpression != "" {
+      t.get{{ $modelName }}sBy{{ $computedIndexName }}ParseFilters(queryInput, input)
+      queryInput.FilterExpression = aws.String(input.FilterExpression) 
+    }
+  {{- end }}
+
   totalRecordsProcessed := int64(0)
   var pageFnErr error
 	pageFn := func(queryOutput *dynamodb.QueryOutput, lastPage bool) bool {
-		if len(queryOutput.Items) == 0 {
-			return false
-		}
+		{{- if gt $lenNonIndexAttributes 0 }}
+      // Only assume an empty page means no more results if there are no filters applied
```

Comment:
Sorry I'm not sure I understand why this is the case, could you explain?

Reply:
When we query a page of records we first fetch them and then apply filters returning whatever is not filtered out as a page of results.  This is true even if we filter out all the records.  It is possible then that we get an empty result set when querying for a page when there are still more results in the table for the given hash/range key.  Previously because filters could not be provided we assumed an empty page meant there were no more results in the table for the given keys

Reply:
Ah okay, that makes sense. could you clarify that in the comment?

Also (doing my best to parse this template code, which is somewhat hard haha) - in the case where we have a filter expression provided and `len(queryOutput.Items) == 0`, should we be returning `!lastPage`. I _think_ right now we are just returning `true` at line 595, after going through some code that wont do anything since there are no `queryOutput.items`